src/wallet/mod.rs

```diff
@@ -356,6 +361,130 @@ impl Wallet {
         Ok(())
     }
 
+    fn create_deposit_op_drivechain_output(
+        sidechain_number: SidechainNumber,
+        sidechain_ctip_amount: Amount,
+        value_sats: Amount,
+    ) -> bdk::bitcoin::TxOut {
+        let deposit_txout =
+            messages::create_m5_deposit_output(sidechain_number, sidechain_ctip_amount, value_sats);
+
+        bdk::bitcoin::TxOut {
+            script_pubkey: bdk::bitcoin::ScriptBuf::from_bytes(
+                deposit_txout.script_pubkey.to_bytes(),
+            ),
+            value: deposit_txout.value.to_sat(),
+        }
+    }
+
+    #[allow(
+        clippy::significant_drop_tightening,
+        reason = "false positive for `bitcoin_wallet`"
+    )]
+    fn create_deposit_psbt(
+        &self,
+        op_drivechain_output: bdk::bitcoin::TxOut,
+        sidechain_address_data: PushBytesBuf,
+        sidechain_ctip: Option<&Ctip>,
+        fee_sats: Option<Amount>,
+    ) -> Result<PartiallySignedTransaction> {
+        let (psbt, _) = {
+            let wallet = self.bitcoin_wallet.lock();
+
+            let mut builder = wallet.build_tx();
+
+            builder
+                // important: the M5 OP_DRIVECHAIN output must come directly before the OP_RETURN sidechain address output.
+                .add_recipient(
+                    op_drivechain_output.script_pubkey,
+                    op_drivechain_output.value,
+                )
+                .add_data(&sidechain_address_data);
+
+            if let Some(fee_sats) = fee_sats {
+                builder.fee_absolute(fee_sats.to_sat());
+            }
+
+            // If the sidechain has a Ctip (i.e. treasury UTXO), the BIP300 rules mandate that we spend the previous
+            // Ctip.
+            if let Some(sidechain_ctip) = sidechain_ctip {
```

Comment:
Is this check needed? Won't all sidechains always have a ctip? My understanding is that you can't deposit to a non-existant sidechain, and a ctip is created when the sidechain is created/activated

Reply:
I believe the very first deposit to a sidechain creates the ctip

https://github.com/LayerTwo-Labs/bip300_bip301_specifications/blob/master/bip300.md
`If a treasury UTXO for this sidechain slot does not exist, then an M5 transaction MUST NOT include an input spending the previous treasury UTXO (as it doesn't exist).`